src/pages/policies/[policyId]/items/new.svelte

```diff
@@ -18,35 +26,40 @@ let open = false
 onMount(() => {
   loadDependents(policyId)
   loadMembersOfPolicy(policyId)
+  loadItems(policyId)
 })
 
 $: metatags.title = formatPageTitle('Items > New')
 
-$: policyId && loadItems(policyId)
-
+$: item.id && $redirect(itemsNewQs(policyId, { itemId: item.id }))
 $: $selectedPolicy.type === PolicyType.Household && !$selectedPolicy.household_id && (open = true)
 
+$: $params.itemId && (item = $selectedPolicyItems.find((i) => i.id === $params.itemId) || {})
 $: breadcrumbLinks = [
   { name: 'Items', url: itemsRoute(policyId) },
-  { name: 'New', url: itemsNew(policyId) },
+  { name: 'New', url: item.id ? itemsNewQs(policyId, { itemId: item.id }) : itemsNew(policyId) },
 ]
 
 const onApply = async (event: CustomEvent) => {
-  item = await addItem(policyId, event.detail)
+  const itemData: NewItemFormData = event.detail
+  await saveOrAddItem(itemData)
   $goto(itemsCheckout(policyId, item.id))
 }
 
 const onSaveForLater = async (event: CustomEvent) => {
-  item.id ? await updateItem(policyId, item.id, event.detail) : (item = await addItem(policyId, event.detail))
+  const itemData: UpdateItemFormData = event.detail
+  saveOrAddItem(itemData)
 
-  if (event.detail.isAutoSaving) {
-    //Todo once autosaving can happen on an empty form send the user immediately to edit.
-    // $goto(itemEdit(policyId, item.id))
-  } else {
+  if (!event.detail.isAutoSaving) {
     $goto(HOME)
   }
 }
 
+const saveOrAddItem = async (itemData: UpdateItemFormData | NewItemFormData) =>
+  item.id
+    ? await updateItem(policyId, item.id, itemData)
+    : (item = await addItem(policyId, itemData as NewItemFormData))
```

Comment:
I had a little trouble reading this inline-function-with-ternary-operator. How do you feel about this alternate syntax, breaking it out with a normal `if` and `return`? I think the `else` didn't need to return anything (so I didn't do so here), but feel free to correct that if I misread it.

```suggestion
const saveOrAddItem = async (itemData: UpdateItemFormData | NewItemFormData) => {
  if (item.id) {
    return updateItem(policyId, item.id, itemData)
  } else {
    item = await addItem(policyId, itemData as NewItemFormData)
  }
}
```